keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -1,20 +1,28 @@
 package com.netflix.spinnaker.keel.api.artifacts
 
-enum class ArtifactType {
-  deb, docker;
-
-  override fun toString() = name
-}
+typealias ArtifactType = String
 
+/**
+ * The release status of an artifact. This may not necessarily be applicable to all
+ * [DeliveryArtifact] sub-classes.
+ */
 enum class ArtifactStatus {
   FINAL, CANDIDATE, SNAPSHOT, RELEASE, UNKNOWN
 }
 
+/**
+ * An artifact as defined in a [DeliveryConfig].
+ *
+ * Unlike other places within Spinnaker, this class does not describe a specific instance of a software artifact
+ * (i.e. the output of a build that is published to an artifact repository), but rather the high-level properties
+ * that allow keel and [ArtifactPublisher] plugins to find/process the actual artifacts.
+ */
 abstract class DeliveryArtifact {
   abstract val name: String
   abstract val type: ArtifactType
   abstract val versioningStrategy: VersioningStrategy
   abstract val reference: String // friendly reference to use within a delivery config
   abstract val deliveryConfigName: String? // the delivery config this artifact is a part of
+  open val statuses: Set<ArtifactStatus> = emptySet()
```

Comment:
Why did you move this to the top level object?

Reply:
Because I figured release status may not forever be a Debian-specific concept.

Reply:
I think we could move those at that future time. I don't think status should be lifted at this moment because it doesn't apply to docker and will be a bit confusing

Reply:
+1 to @emjburns's point.

Reply:
I added a `@JsonIgnore` to `DockerArtifact`, but will keep it in the base class.

Reply:
hmm, fair point. Thanks for adding a `@JsonIgnore` to the docker artifact.